packages/wallet/api/src/sharingUnserializer.js

```diff
@@ -0,0 +1,81 @@
+// @ts-check
+import { Far, makeMarshal } from '@endo/marshal';
+
+/**
+ * Make a pair of unserializers that share low-privilege values.
+ *
+ * The "high" privilege unserializer can understand objects as slots, such as
+ * those from the wallet serializer. The "low" privilege unserializer can only
+ * understand strings as slots, such as those from the board serializer.
+ *
+ * This provides a way, for instance, to get identical brand references from
+ * both a public contract and a user's purses.
+ *
+ * @param {string} [lowProperty] - The key name to check on a high privilege
+ * slot object. If the string value at this key is equivalent to the value of
+ * another low privilege slot, the high privilege unserializer will return the
+ * same object for both slots.
+ */
+export const makeSharingUnserializer = lowProperty => {
+  assert(lowProperty, 'lowProperty must be specified for sharing unserializer');
+  const seen = {
+    // high: new WeakMap(),  // ISSUE: WeakMap conflicts with string keys
```

Comment:
is that ISSUE one that can be resolved? if not, I'd expect to see a comment like "Ideally this would be a weak map for `<reasons>` but that doesn't handle string keys"